cblite/llm/Gemini.cc

```diff
@@ -22,25 +22,69 @@ using namespace std;
 using namespace fleece;
 using namespace litecore;
 
-alloc_slice Gemini::run(Value rawSrcPropValue, const string& modelName) {
-    string restBody = format("{\"model\": \"models/%s\", \"content\": {\"parts\": [{\"text\": \"%.*s\"}]}}", modelName.c_str(), SPLAT(rawSrcPropValue.asString()));
+vector<alloc_slice> Gemini::run(const string& modelName, vector<Value> propVec) {
+    int batchSize = 50, docRemainder = propVec.size() % batchSize;
+    long batches = propVec.size() / batchSize;
+    if (propVec.size() % batchSize != 0) {
+        batches++;
```

Comment:
This is usually accomplished by doing a sort of "ceiling" in one line:

`long batches = (propVec.size() + BATCH_SIZE - 1) / BATCH_SIZE`